packages/@expo/cli/src/start/server/metro/MetroBundlerDevServer.ts

```diff
@@ -131,8 +133,30 @@ export class MetroBundlerDevServer extends BundlerDevServer {
         path.relative(appDir, filepath.replace(/\.[tj]sx?$/, '.js'))
       );
       if (contents) {
+        let src = contents.src;
+        if (includeSourceMaps && contents.map) {
+          const artifactBasename = encodeURIComponent(path.basename(artifactFilename) + '.map');
+          src = src.replace(
+            /\/\/# sourceMappingURL=.*/g,
+            `//# sourceMappingURL=${artifactBasename}`
+          );
+          files.set(artifactFilename + '.map', {
+            contents: JSON.stringify({
+              version: contents.map.version,
+              sources: contents.map.sources.map((source: string) => {
+                return typeof source === 'string' && source.startsWith(this.projectRoot)
+                  ? path.relative(this.projectRoot, source)
+                  : source;
+              }),
+              sourcesContent: new Array(contents.map.sources.length).fill(null),
+              names: contents.map.names,
+              mappings: contents.map.mappings,
+            }),
```

Comment:
This is something I'd ideally would want to get rid of 🤔 

The problem here is that we need `sources` to be absolute for `source-map-support` (at least, as far as I know), but the output needs to be relative to the project root again.

It's potentially possible to see whether we can change the output to contain the absolute path to `sourceRoot`, but it's unclear whether `sourceRoot` will be ignored on deployed platforms.